tests/ExponentialBackoffTest.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace ipl\Tests\Stdlib;
+
+use Exception;
+use ipl\Stdlib\ExponentialBackoff;
+use LogicException;
+
+class ExponentialBackoffTest extends \PHPUnit\Framework\TestCase
+{
+    public function testInvalidMaxWaitTime()
+    {
+        $this->expectException(LogicException::class);
+        $this->expectExceptionMessage('Max must be larger than min');
+
+        new ExponentialBackoff(1, 500, 100);
+    }
+
+    public function testMinAndMaxWaitTime()
+    {
+        $backoff = new ExponentialBackoff();
+        $this->assertSame(100, $backoff->getMin());
+        $this->assertSame(10 * 1000, $backoff->getMax());
+
+        $backoff
+            ->setMin(200)
+            ->setMax(500);
+
+        $this->assertSame(200, $backoff->getMin());
+        $this->assertSame(500, $backoff->getMax());
+    }
+
+    public function testRetriesSetCorrectly()
+    {
+        $backoff = new ExponentialBackoff();
+
+        $this->assertSame(1, $backoff->getRetries());
+        $this->assertSame(5, $backoff->setRetries(5)->getRetries());
+        $this->assertNotSame(10, $backoff->setRetries(5)->getRetries());
+    }
+
+    public function testGetWaitTime()
```

Comment:
Should also test the waiting time for attempt `n`, where `0..n-1` has not yet been called.